pkg/xen-tools/initrd/chroot2.c

```diff
@@ -52,17 +57,33 @@ int main(int argc, char **argv) {
         .workdir = argv[2],
         .uid = uid,
         .gid = gid,
-        .command = argv[5],
-        .args = argv + 5,
+        .command = argv[6],
+        .args = argv + 6,
     };
-
     child_pid = clone(childFunc, child_stack + STACK_SIZE,
                       CLONE_NEWPID | SIGCHLD, &args);
     if (child_pid < 0) {
         perror("clone() failed:");
         return -1;
     }
 
+    /*
+     * Open a file and write a PID of the child process in order
+     * to do attach to its namespace.
+     */
+    fd = open(pid_file, O_CREAT|O_WRONLY|O_TRUNC, 0600);
+    if (fd < 0) {
+        /* Don't consider as fatal */
+        perror("open(pid_file) failed:");
+    } else {
+        char buf[64];
+        int len;
+
+        len = snprintf(buf, sizeof(buf), "%u\n", child_pid);
+        write(fd, buf, len);
```

Comment:
May be I would add an error handing here, but it's not critical.

Reply:
I would like to keep my version.